eng/scripts/TypeSpec-Requirement.ps1

```diff
@@ -0,0 +1,75 @@
+[CmdletBinding()]
+param (
+)
+Set-StrictMode -Version 3
+
+. $PSScriptRoot/ChangedFiles-Functions.ps1
+. $PSScriptRoot/Logging-Functions.ps1
+
+$repoPath = Resolve-Path "$PSScriptRoot/../.."
+$pathsWithErrors = @()
+
+$filesToCheck = @(Get-ChangedSwaggerFiles)
+if (!$filesToCheck) {
+  LogInfo "No swagger files found to check"
+}
+else {
+  # Example: specification/foo/resource-manager/Microsoft.Foo/stable/2023-01-01/Foo.json
+  # - Forward slashes on both Linux and Windows
+  foreach ($file in $filesToCheck) {
+    LogInfo "Checking $file"
+
+    $jsonContent = Get-Content (Join-Path $repoPath $file) | ConvertFrom-Json -AsHashtable
+
+    if ($null -ne ${jsonContent}?["info"]?["x-typespec-generated"]) {
+      LogInfo "  Swagger was generated from TypeSpec (contains '/info/x-typespec-generated')"
+      # Skip further checks, since spec is already using TypeSpec
+      continue
+    }
+    else {
+      LogInfo "  Swagger was not generated from TypeSpec (missing '/info/x-typespec-generated')"
+    }
+
+    # Example: specification/foo/resource-manager/Microsoft.Foo
+    $pathToServiceName = ($file -split '/')[0..3] -join '/'
+
+    $urlToStableFolder = "https://github.com/Azure/azure-rest-api-specs/tree/main/$pathToServiceName/stable"
```

Comment:
What about `azure-rest-api-specs-pr` repo? (note the additional `-pr`)

Reply:
The check should specifically look at public main, not pr main.  The goal is to determine if the spec has ever released a public stable version, which only applies to public main, not any other repo or branch.